### PR TITLE
513 project not social mediasso friendly (draft)

### PR DESCRIPTION
**Pull request by chao-mu**

### CODE REVIEW

File: .github/workflows/test-build.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+name: Lint
```
[the remark this answers is not in the document]
Comment (chao-mu) [reply .github/workflows/test-build.yml:1]: I was thinking about that. My thought was that perhaps code duplication is worth the benefit of clearness of reporting - i.e. there is a separate line item for tasks. Downside is of course time. Sort of an interesting piece here is that I believe the linting rules run during the build! I am on the fence.

File: frontend/src/profile/progress/ProgressHistory.tsx [hunk 2]
```diff
@@ -295,7 +295,7 @@ const ProgressHistory: React.FC<ProgressHistoryProps> = ({
 }) => {
     const { user } = useAuth();
     const api = useApi();
-    const request = useRequest();
+    const request = useRequest<unknown>();
```
[the remark this answers is not in the document]
Comment (chao-mu) [reply frontend/src/profile/progress/ProgressHistory.tsx]: The actual type is AxiosResponse<User, any> and I was avoiding the any without putting much thought into it. Apparently the any is the default value for the second type parameter and I can declare it without using 'any'. Fixed.

File: .github/workflows/test-build.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (chao-mu) [reply .github/workflows/test-build.yml]: Okay, decided to combine them. We'll want to eventually expand the github action to lint and build all parts of the code base (presumably - ideally what's changed, but that's a story for a different day), so should probably combine them while it's still simple.

File: frontend/tsconfig.json [hunk 3]
```diff
@@ -1,39 +1,41 @@
...
+  },
+  "include": [
+    "src",
+    "cypress.config.ts",
+    "vitest.config.ts",
+    ".eslintrc.cjs",
+    "scripts",
+    "./*.cjs",
+    "./*.mjs",
+    "./build/types/**/*.ts",
```
Comment (jackstenglein) [inline frontend/tsconfig.json:34]: Why are we including the build directory?

**Pull request merged**